[wasm] Set WasmEnableWebcil to true by default

lambdageek opened the pull request:
Switch to webcil by default in WebAssembly.

Contributes to https://github.com/dotnet/runtime/issues/80807

Also fix some debugger test failures.

lambdageek committed:
[wasm] Set WasmEnableWebcil to true by default

teo-tsirpanis commented on docs/design/mono/webcil.md:
```diff
@@ -2,21 +2,83 @@
+
+That is, the module imports linear memory 0 and exports:
+* a global `i32` `webcilVersion` encoding the version of the WebAssembly wrapper (currently 0),
+* a function `getWebcilSize : i32 -> ()` that writes the size of the Webcil payload to the specified
+  address in linear memory as a `u32` (that is: 4 LE bytes).
```
Why not just return the size?

teo-tsirpanis replied:
Thanks, I just saw the rationale sections below. 😅

lambdageek commented:
the symbol server debugger tests are failing with a pdb checksum mismatch.  which is weird because we're apparently reading the debug directory data correctly.  but somehow it is wrong

lambdageek committed:
WebcilReader: copy updated struct for BlobReader.ReadUtf8NullTerminated

When using reflection to call a method on BlobReader, make sure to
copy the struct after the call since the internal state is modified

Fixes PdbChecksum extraction from Webcil files

(This was a problem even before webcil-in-wasm)

lambdageek commented:
/azp run runtime-wasm

lambdageek commented:
@maraf @radical this is ready for review. the multi-threaded debugger issues are known issues. single-threaded debugging is passing with webcil-in-wasm now

maraf reviewed (approved):
Looks good to me 👍

lambdageek commented:
@maraf will work on downstream test fixes for dotnet/sdk and dotnet/aspnetcore.  We're going to merge this PR.
